src/server.c

```diff
@@ -1499,7 +1498,7 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
     if (hasActiveChildProcess() || ldbPendingChildren())
     {
         run_with_period(1000) receiveChildInfo();
-        checkChildrenDone();
+        handleCompletedChildren();
```

Comment:
Please note that `serverCron()` was called server.hz times per second, but `replicationCron()` is 1 times per second, which means there may be a second delay.